public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AssociateExisting/containers/AssociateExisting.tsx

```diff
@@ -0,0 +1,282 @@
+
+    return existingDetectorsAvailableToAssociate.map((detector) => ({
+      label: detector.name,
+    }));
+  }, [existingDetectorsAvailableToAssociate]);
+
+  return (
+    <div className="associate-existing">
+      <EuiText size="xs">
+        <p>
+          View existing anomaly detectors across your system and add the
+          detector(s) to a dashboard and visualization.{' '}
+          <a href={`${BASE_DOCS_LINK}/ad`} target="_blank">
+            Learn more <EuiIcon type="popout" />
+          </a>
+        </p>
+      </EuiText>
+      <EuiSpacer size="l" />
+      <EuiTitle size="s">
+        <h3>Select detector to associate</h3>
+      </EuiTitle>
+      <EuiSpacer size="m" />
+      <EuiText size="xs" color="subdued" style={{ paddingBottom: '3px' }}>
+        Eligible detectors don't include high-cardinality detectors.
+      </EuiText>
+      {!existingDetectorsAvailableToAssociate && <EuiLoadingSpinner size="l" />}
+      {existingDetectorsAvailableToAssociate && (
+        <EuiComboBox
+          isLoading={isLoading}
+          id="associate-existing__select"
+          options={options}
+          selectedOptions={selectedOptions}
+          onChange={(selectedOptions) => {
+            let detector = {} as DetectorListItem | undefined;
+
+            if (selectedOptions && selectedOptions.length) {
+              const match = existingDetectorsAvailableToAssociate.find(
+                (detector) => detector.name === selectedOptions[0].label
+              );
+              detector = match;
+            }
+            associateExistingProps.setSelectedDetector(detector);
+          }}
+          aria-label="Select an anomaly detector to associate"
+          isClearable
+          singleSelection
+          placeholder="Search for an anomaly detector"
+        />
+      )}
+      <EuiSpacer size="xl" />
+      {detector && (
+        <>
+          <EuiFlexGroup justifyContent="spaceBetween" alignItems="flexStart">
+            <EuiFlexItem>
+              <EuiText>
+                <h4>{detector.name}</h4>
+              </EuiText>
+              <EuiSpacer size="s" />
+              <EuiHealth color={stateToColorMap.get(detector.curState)}>
+                {renderTime(detector.enabledTime)}
+              </EuiHealth>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiLink href={`${PLUGIN_NAME}#/detectors/${detector.id}`}>
+                View detector page
+              </EuiLink>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiHorizontalRule margin="m" />
+          <ul className="associate-existing__detector-details">
+            {[
+              ['Indexes', (detector) => detector.indices],
+              [
+                'Anomalies last 24 hours',
+                (detector) => detector.totalAnomalies,
+              ],
+              [
+                'Last real-time occurence',
+                (detector) => renderTime(detector.lastActiveAnomaly),
```

Comment:
just to confirm, if this is undefined/null, will it return an empty string?

Reply:
it returns `-` as defined by `DEFAULT_EMPTY_DATA`, also what UX wanted to be displayed here. Photo in description shows example.